pkg/minikube/cluster/cluster.go

```diff
@@ -182,31 +182,137 @@ func StartCluster(h sshAble) error {
 	return nil
 }
 
-type fileToCopy struct {
+type CopyableFile interface {
+	io.Reader
+	GetLength() int
+	GetAssetName() string
+	GetTargetDir() string
+	GetTargetName() string
+	GetPermissions() string
+}
+
+type BaseAsset struct {
+	Data        []byte
+	Reader      io.Reader
+	Length      int
 	AssetName   string
 	TargetDir   string
 	TargetName  string
 	Permissions string
 }
 
-var assets = []fileToCopy{
-	{
-		AssetName:   "deploy/iso/addon-manager.yaml",
-		TargetDir:   "/etc/kubernetes/manifests/",
-		TargetName:  "addon-manager.yaml",
-		Permissions: "0640",
+func (b *BaseAsset) GetAssetName() string {
+	return b.AssetName
+}
+
+func (b *BaseAsset) GetTargetDir() string {
+	return b.TargetDir
+}
+
+func (b *BaseAsset) GetTargetName() string {
+	return b.TargetName
+}
+
+func (b *BaseAsset) GetPermissions() string {
+	return b.Permissions
+}
+
+type FileAsset struct {
+	BaseAsset
+}
+
+func (f *FileAsset) loadData() error {
+	contents, err := ioutil.ReadFile(f.AssetName)
+	if err != nil {
+		return err
+	}
+	f.Data = contents
+	f.Length = len(contents)
+	f.Reader = bytes.NewReader(f.Data)
```

Comment:
Ah nevermind, I see you use it to implement Read below. What if you make Reader and Data both private? That would make the usage here more clear.